bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/internal/p2/engine/SimpleProfileRegistry.java

```diff
@@ -604,25 +604,15 @@ private void saveProfile(Profile profile) {
 
 		profile.setTimestamp(currentTimestamp);
 		profile.setChanged(false);
-		OutputStream os = null;
-		try {
-			if (shouldGzipFile)
-				os = new BufferedOutputStream(new GZIPOutputStream(new FileOutputStream(profileFile)));
-			else
-				os = new BufferedOutputStream(new FileOutputStream(profileFile));
+		try (OutputStream os = shouldGzipFile
+				? new BufferedOutputStream(new GZIPOutputStream(new FileOutputStream(profileFile)))
+				: new BufferedOutputStream(new FileOutputStream(profileFile))) {
```

Comment:
```suggestion
				? new GZIPOutputStream(new FileOutputStream(profileFile))
				: new FileOutputStream(profileFile)) {
```
The `XMLWriter` super class already wraps the passed `OutputStream` into a `BufferedWriter`, so this additional buffer is actually just extra copy work.